mx.py

```diff
@@ -7053,6 +7053,43 @@ def run_java(args, nonZeroIsFatal=True, out=None, err=None, cwd=None, timeout=No
         jdk = get_jdk()
     return jdk.run_java(args, nonZeroIsFatal=nonZeroIsFatal, out=out, err=err, cwd=cwd, timeout=timeout, env=env, addDefaultArgs=addDefaultArgs)
 
+def run_java_min_heap(args, benchName='# MinHeap:', overheadFactor=1.5, referenceHeap=4096, out=None, err=None, cwd=None, timeout=None, env=None, addDefaultArgs=True, jdk=None):
+    """computes the minimum heap size required to run a Java program within a certain overhead factor"""
+
+    def run_with_heap(heap, args, timeout=timeout, suppressStderr=True, nonZeroIsFatal=False):
+        log('Trying with ' + str(heap) + 'MB of heap...')
+        with open(os.devnull, 'w') as fnull:
+            exitCode = run_java(['-Xmx' + str(heap) + 'M'] + args, nonZeroIsFatal=nonZeroIsFatal, out=out, err=fnull if suppressStderr else err, cwd=cwd, timeout=timeout, env=env, addDefaultArgs=addDefaultArgs)
+            if exitCode:
+                log('failed')
+            else:
+                log('succeeded')
+            return exitCode
+
+    if overheadFactor > 0:
+        t = time.time()
+        if run_with_heap(referenceHeap, args, timeout=timeout, suppressStderr=False):
+            return 1
+        referenceTime = time.time() - t
+        maxTime = referenceTime * overheadFactor
+        log('Reference time = ' + str(referenceTime))
+        log('Maximum time = ' + str(maxTime))
+    else:
+        maxTime = None
+
+    heap = 10
+    while run_with_heap(heap, args, timeout=maxTime):
```

Comment:
Do you want to start with some reasonable heap size and then binary search your way down to near the minimum, rather than (as I read it) starting with a heap size that's too small (10MB?) and growing the heap by a fixed amount (10MB then 1MB?) after each failure?  How many iterations does it typically take you to find the minimum?  How does that number of iterations compare with log2(referenceHeap)?  How long does each iteration take for benchmarks you care about?